### PR TITLE
{dev,net}/ieee802154_*: add initial Kconfig modeling

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR add the initial Kconfig modeling for the lower layers of IEEE 802.15.4. This includes:
- Adding feature symbols to indicate presence of Radio HAL or Netdev Driver based driver.
- Modeling of SubMAC and `netdev_ieee802154_submac`
- Adding `tests/ieee802154_submac` to the list of Kconfig test executed by Murdock

This PR also refactors the IEEE 802.15.4 menu, because `MODULE_IEEE802154` just provides helper functions for processing IEEE 802.15.4 frames. Last but not least, it removes unused configurations in ` nrf802154`.

### Testing procedure
Murdock should be enough I guess.

### Issues/PRs references
Depends on #16837

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
I think this needs a rebase before review

**Comment by fjmolinas** [comment]
@jia200x this needs a rebase now!

**Comment by jia200x** [comment]
is there anything missing here?

**Comment by fjmolinas** [comment]
> is there anything missing here?

Yes, I have some unanswered comments, I think currently the submac test does not build

File: cpu/nrf52/radio/nrf802154/Kconfig [hunk 1]
```diff
@@ -8,5 +8,6 @@
 config HAS_RADIO_NRF802154
     bool
     select HAVE_NRF5X_RADIO
+    select MODULE_NRF802154 if MODULE_NETDEV_DEFAULT
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply cpu/nrf52/radio/nrf802154/Kconfig]: I think this cannot actually select the module here.

As `MODULE_NRF802154` is part of a choice I added that to be the default choice.   If the bluetooth stack is used then that choice can be overridden to select the BLE one first... I think.

File: sys/net/link_layer/ieee802154/Kconfig [hunk 2]
```diff
@@ -4,26 +4,38 @@
...
     help
         IEEE 802.15.4 security interface
 
 config MODULE_IEEE802154_SUBMAC
-    bool "IEEE 802.15.4 submac"
+    bool "IEEE 802.15.4 SubMAC"
     select MODULE_XTIMER
+    select MODULE_RANDOM
+    depends on FEATURE_IEEE802154
+    depends on TEST_KCONFIG
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/net/link_layer/ieee802154/Kconfig]: Done

File: sys/net/link_layer/ieee802154/Kconfig [hunk 3]
```diff
@@ -4,26 +4,38 @@
...
+config MODULE_IEEE802154
+    bool "IEEE 802.15.4 helpers"
+    help
+        Provide IEEE 802.15.4 helpers
 
 config MODULE_IEEE802154_SECURITY
     bool "IEEE 802.15.4 security"
     select MODULE_CRYPTO
     select MODULE_CIPHER_MODES
+    depends on TEST_KCONFIG
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/net/link_layer/ieee802154/Kconfig]: Done

File: sys/net/link_layer/ieee802154/Kconfig [hunk 4]
```diff
@@ -4,26 +4,38 @@
 # General Public License v2.1. See the file LICENSE in the top level
 # directory for more details.
 
-menuconfig MODULE_IEEE802154
-    bool "IEEE 802.15.4 support"
+menuconfig FEATURE_IEEE802154
+    bool "Enable IEEE 802.15.4 support"
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/net/link_layer/ieee802154/Kconfig]: I will leave that to @jia200x

File: sys/net/link_layer/ieee802154/Kconfig [hunk 5]
```diff
@@ -4,26 +4,38 @@
...
     depends on TEST_KCONFIG
+    default y if HAVE_IEEE802154_RADIO && MODULE_NETDEV_DEFAULT
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/net/link_layer/ieee802154/Kconfig:10]: done

File: sys/net/link_layer/ieee802154/Kconfig [hunk 6]
```diff
@@ -4,26 +4,38 @@
 # General Public License v2.1. See the file LICENSE in the top level
 # directory for more details.
 
-menuconfig MODULE_IEEE802154
-    bool "IEEE 802.15.4 support"
+menuconfig FEATURE_IEEE802154
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/net/link_layer/ieee802154/Kconfig]: done, though I think we should rename the module to `IEEE802154_HELPERS`.  A networking newbie like myself can get easily confused,,,

File: drivers/Kconfig.net [hunk 7]
```diff
@@ -21,4 +21,36 @@ rsource "slipdev/Kconfig"
...
+
+config HAVE_IEEE802154_RADIO_HAL
+    bool
+    select HAVE_IEEE802154_RADIO
+    help
+        At least one device provides an IEEE 802.15.4 Radio HAL implementation
+
+config HAVE_NETDEV_IEEE802154
+    select HAVE_IEEE802154_RADIO
+    bool
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply drivers/Kconfig.net]: done

File: drivers/Kconfig.net [hunk 8]
```diff
@@ -21,4 +21,36 @@ rsource "slipdev/Kconfig"
 rsource "sx126x/Kconfig"
 rsource "sx127x/Kconfig"
 rsource "tja1042/Kconfig"
+
+config MODULE_NETDEV_IEEE802154
+    bool
+    help
+        Netdev IEEE 802.15.4 module, provides common functionalities for
+        netdev based IEEE 802.15.4 devices.
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply drivers/Kconfig.net:30]: Things get a bit spread out.  I will have to take a closer look what belongs where.

**Comment by fjmolinas** [comment]
Still some issues @MrKevinWeiss (assuming you are taking over)

**Comment by MrKevinWeiss** [comment]
I think @leandrolanzieri is working on a broader solution that will make this obsolete...

**Comment by Teufelchen1** [comment]
@MrKevinWeiss this now outdated, right? Can it be closed?

**Comment by MrKevinWeiss** [comment]
Yup

**Pull request closed without merging**